erpnext/selling/doctype/quotation/quotation.py

```diff
@@ -68,13 +68,19 @@ def update_opportunity_status(self, opportunity=None):
 
 	def declare_enquiry_lost(self, lost_reasons_list, detailed_reason=None):
 		if not self.has_sales_order():
+			get_lost_reasons = frappe.get_list('Opportunity Lost Reason',
+			fields = ["name"])
+			lost_reasons_lst = [reason.get('name') for reason in get_lost_reasons]
 			frappe.db.set(self, 'status', 'Lost')
 
 			if detailed_reason:
 				frappe.db.set(self, 'order_lost_reason', detailed_reason)
 
 			for reason in lost_reasons_list:
-				self.append('lost_reasons', reason)
+				if reason.get('lost_reason') in lost_reasons_lst:
+					self.append('lost_reasons', reason)
+				else:
+					frappe.throw(_("Invalid lost reason <b>{0}</b>, please create a new lost reason".format(reason.get('lost_reason'))))
```

Comment:
@lokesh-indictrans This string won't be translated properly since it has html tags inside it. Use `frappe.bold()` function instead of HTML tags.

Reply:
@deepeshgarg007  Updated code for requested changes.